### ISSUE TITLE
DUNNET - adventure game

### ISSUE THREAD

**Issue by larsbrinkhoff** [PDP-10/its#146]
I asked Ron Schnell for a copy of the MACLISP version of DUNNET.  We'll see how that goes.

**Comment by eswenson1** [comment]
https://opensource.apple.com/source/emacs/emacs-51/emacs/lisp/play/dunnet.el

**Comment by larsbrinkhoff** [comment]
Emacs Lisp is rather close to MacLisp, so I suppose it wouldn't be that much work to get it running.

That was not what I had in mind, but now that you mention it, maybe there are other useful Emacs Lisp programs that could be made to work in ITS.

**Comment by eswenson1** [comment]
I assume it requires a data file though, so we'd have to find that.  I've actually never heard of DUNNET, although I was an ADVENT and ZORK (DUNGEON) enthusiast back in the day.

**Comment by larsbrinkhoff** [comment]
The Emacs version of Dunnet is distributed with GNU Emacs:  
http://git.savannah.gnu.org/cgit/emacs.git/tree/lisp/play/dunnet.el

I think the data is inside the Lisp file.

But I was really hoping that Ron would have a copy of the MacLisp DUNNET, and that he would let us have it.

**Comment by eswenson1** [comment]
Yes. I agree.  Let's see what Ron says.

**Comment by larsbrinkhoff** [comment]
Ron Schnell and @nessus42 say they don't have it.  It could be on backup tapes from MIT-EECS.

**Comment by nessus42** [comment]
I highly doubt that MIT EECS has any backup tapes left lying around from the early '80s. And in the unlikely event that they did have them, would they be readable after all these years? And even if they were readable, wouldn't you need Tops 20 to read them? (EECS didn't have any ITS computers. Only Tops 20 ran on its PDP 10.)

**Comment by larsbrinkhoff** [comment]
You're probably right.  I just wanted to record the information I got from Ron Schnell.

MIT do have some people working on archiving their backup tapes, but it's highly uncertain which have been saved, and what state they're in.

But if we're lucky, we have a Unix tool to read TOPS-20 backup tapes in DUMPER format.  Or, failing that, there are people running TOPS-20 on real and emulated machines.

**Comment by larsbrinkhoff** [comment]
@Devon7 should know if there are any MIT-EECS backup tapes.

**Comment by nessus42** [comment]
I doubt it. I stopped working at the MIT EECS facility in 1989 or 1990. By that time the Decsystem 20 was long gone. Teaching had moved onto some kind of HP workstations that had Scheme development environments on them, and administration had moved completely to Vaxstations running BSD. I don't even know if we had a working 9-track tape drive anymore.

**Comment by Devon7** [comment]
I roamed the Courts of Chaos building & testing CADR memory cards, I probably logged into MIT-Eddie, I imagine undergrad interns spun tapes and no doubt there was a small rack in the machine room but I've no knowledge of nor contact with EECS backups.

			Peace
				—Devon

P.S. Hi Nessus!  Going to Arisia this weekend?  I’ll probably give it a miss due to weekend classes.

**Comment by larsbrinkhoff** [comment]
There may have been a port of dunnet.el to Maclisp, by @k-stailey:  
http://its.svensson.org/KES%3bDUNNET%201

**Comment by larsbrinkhoff** [comment]
@cpryland was a sysadmin at MIT-EECS.

**Comment by larsbrinkhoff** [comment]
The stakes are getting high.  Now there's beverage involved!

> I will buy many beers if anyone can find me the original version. It was much different! Maybe better. Instead of finding yourself in a Vax/UNIX, you were inside a Dec-20 running Twenex. It used the COMND library, which simulated the COMND JRST. I wrote the elisp version trying to remember the "sort" of stuff I did with the original version, but even I would be surprised by what was actually in there...

**Comment by larsbrinkhoff** [comment]
Brad Parker found a copy somewhere at MIT!  I don't know how or where.

**Comment by Devon7** [comment]
So that’s why he asked me to re-activate his account — to access our ToTS archive.

		Peace
			—Devon

P.S. I’m missing some essential part of this conversation - what is it exactly that Brad found?

**Comment by larsbrinkhoff** [comment]
Brad found Ron Schnell's user directory (I don't know from which machine) with an early version of his game DUNNET.

**Comment by Alexey-Slyusar** [comment]
@larsbrinkhoff could you please tell what is the current situation with the running DUNNET on ITS. I mean that I didn't find the game in ITS main branch and it didn't mentioned in `games.md`.
Do you have the DUNNET MacLisp code?

**Comment by larsbrinkhoff** [comment]
The code is here: https://github.com/Quogic/DunnetPredecessor

It's written for TOPS-20, and uses a library to access the command line processor on that system.  I have written a somewhat weak work-alike.  However, the DUNNET code is also rather incomplete, so there's not much of a game.

**Comment by Alexey-Slyusar** [comment]
Lars, thank you for the commentary.
The library that you mentioned is COMRED I guess.
Could you please share with me the work-alike code that you've written. It will be good to learn and tinkering with your code for a very beginner MacLisp programmer / hobbyist like I am, I think. It is interesting and good from an educational point of view.

**Comment by eswenson1** [comment]
I thought he was talking about the COMRED that he wrote -- LIBDOC; COMRED 1 in DB.

**Comment by Alexey-Slyusar** [comment]
Eric, thank you for the reference. I'll look at the `LIBDOC; COMRED 1 in DB.`

**Comment by Alexey-Slyusar** [comment]
There is `LIBDOC; COMRED 1` on ES also, though. ;-)

**Comment by eswenson1** [comment]
Right. Same version.  I've try to keep ES refreshed with the latest DB changes.
